src/components/Modeler.vue

```diff
@@ -614,12 +614,12 @@ export default {
     },
     async renderPaper() {
       await this.$nextTick();
-      this.paper.freeze();
+      this.paperManager.freezePaper();
       this.isRendering = true;
       await this.waitForCursorToChange();
-      this.paper.once('render:done', () => this.isRendering = false);
+      this.paperManager.addOnceHandler('render:done', () => this.isRendering = false);
       this.parse();
-      this.paper.unfreeze();
+      this.paperManager.unfreezePaper();
```

Comment:
I tried to convert this code to use the `performAtomicAction()` method but I couldn't make it work. Something about the async/await behaviour here screwed this up. This is something that can be tackled later.

Reply:
@JCook21 I forgot to include this in my review. You can make `performAtomicAction` an async function and `await` the callback:
```js
async performAtomicAction(callback) {
  this._paper.freeze();
  await callback(this._paper);
  this._paper.unfreeze();
}
```

Then here you can have:
```js
performAtomicAction(async () => {
  this.isRendering = true;
  await this.waitForCursorToChange();
  this.paperManager.addOnceHandler('render:done', () => this.isRendering = false);
  this.parse();
});
```

Reply:
Let me try it. I tried something like this earlier but it didn't work. I probably missed something...